app/jobs/create_institution_records_job.rb

```diff
@@ -16,7 +16,9 @@ def perform(institution_type_id)
       page += 1
 
       batch_insert_institutions = []
-      raw_institutions.each_with_object({}) do |raw_institution, institution|
+      raw_institutions.each do |raw_institution|
```

Comment:
I think that it can be left as is. Doing refactoring to save one or two lines of code is sort of missing the point in my opinion. Using fancy iteration techniqes like each_with_object later leads to bugs like this. Trying to refactor away two lines of perfectly readible code has cost me two hours of debugging later. I would prefer to use simple iteration that is easy to understand and where it is hard to make mistakes like that lead to bugs later.

Reply:
Man, `map` is not fancy :D If you want to write ruby, you need to know `map`, `each_with_object`, `reduce`, and other `Enumerable` methods by heart. There's no other way around it. If you were to use loops, you might as well write this in C. The whole power of Ruby comes from being able to express the very essence of what you're doing in a concise, readable way. Once you get some experience using those higher-order functions, you'll realize it's actually way easier to write and way harder to make mistakes using it - but there's no other way to gain that experience than just to use them.

Reply:
Look at this this way, your original code was basically this:
```ruby
result = []
i = 0
loop do
  result << transform(data[i])
  break unless data[i]
end
```
The map version is basically this:
```ruby
result = data.map { |obj| transform(obj) }
```
In the first version, you have to create 2 variables, which can lead to mistakes. You have to write a break condition, which can lead to mistakes. You have to mutate an object, which can lead to mistakes. You need to do some thinking to understand the content of the result object at any point in time.
In the second version, there are no intermediate variables, no conditions, no mutations. You immediately see that the result object is an array of the same length as the data array. Sure, you need to understand how `map` works, but this is fundamental in any higher-order language.

Reply:
The code logic you mention was used when we needed the index to access the hash inside the hash of hashes with stringified index as second key. After they changed it to array of hashes I switched to the logic that is used now.

Reply:
I'm not sure I understand. I don't want to block the development, so if you don't know how to use map, we can leave it is as it is and schedule a quick pair programming to fix it next week.